AmazonChimeSDKDemo/AmazonChimeSDKDemo/VideoModel.swift

```diff
@@ -347,29 +367,69 @@ class VideoModel: NSObject {
+                    continue
+                }
+                updatedSources[source] = pendingRemoteVideoSourceConfigurations[source]
+                videoCount += 1
+                if videoCount == remoteVideoTileCountPerPage {
+                    break
+                }
+            }
+        }
+
         audioVideoFacade.updateVideoSourceSubscriptions(addedOrUpdated: updatedSources, removed: [])
     }
     
-    func addContentShareVideoSource(attendeeId : String) {
+    func addContentShareVideoSource() {
         var updatedSources:[RemoteVideoSource: VideoSubscriptionConfiguration] = [:]
+
+        for remoteVideoSource in pendingRemoteVideoSourceConfigurations {
+            if isAttendeeIdContentShare(attendeeId: remoteVideoSource.key.attendeeId) {
+                moveVideoSourceFromPendingToNormalByAttendeeId(attendeeId: remoteVideoSource.key.attendeeId)
+            }
+        }
+
         for remoteVideoSource in remoteVideoSourceConfigurations {
-            if(remoteVideoSource.key.attendeeId == attendeeId) {
+            if isAttendeeIdContentShare(attendeeId: remoteVideoSource.key.attendeeId) {
                 updatedSources[remoteVideoSource.key] = remoteVideoSourceConfigurations[remoteVideoSource.key]
-                
             }
         }
         audioVideoFacade.updateVideoSourceSubscriptions(addedOrUpdated: updatedSources, removed: [])
     }
 
+    func removeContentShareVideoSources() {
+        var removedSources:[RemoteVideoSource] = []
+        for remoteVideoSource in remoteVideoSourceConfigurations {
+            if isAttendeeIdContentShare(attendeeId: remoteVideoSource.key.attendeeId) {
+                removedSources.append(remoteVideoSource.key)
+            }
+        }
+        audioVideoFacade.updateVideoSourceSubscriptions(addedOrUpdated: [:], removed: removedSources)
+    }
+
+    func removeNonContentShareVideoSources() {
+        var removedSources:[RemoteVideoSource] = []
+        for remoteVideoSource in remoteVideoSourceConfigurations {
+            if !isAttendeeIdContentShare(attendeeId: remoteVideoSource.key.attendeeId) {
+                removedSources.append(remoteVideoSource.key)
+            }
+        }
+        audioVideoFacade.updateVideoSourceSubscriptions(addedOrUpdated: [:], removed: removedSources)
+    }
+
     func pauseAllRemoteVideos() {
         for remoteVideoTileState in remoteVideoTileStates {
             audioVideoFacade.pauseRemoteVideoTile(tileId: remoteVideoTileState.0)
         }
     }
 
     func unsubscribeAllRemoteVideos() {
-        let remoteVideoSources: [RemoteVideoSource] = getRemoteVideoSubscriptionsFromRemoteVideoTileStates(remoteVideoTileStates: remoteVideoTileStates)
-        audioVideoFacade.updateVideoSourceSubscriptions(addedOrUpdated:[:], removed:remoteVideoSources)
+        audioVideoFacade.updateVideoSourceSubscriptions(addedOrUpdated:[:], removed: Array(remoteVideoSourceConfigurations.keys))
```

Comment:
Ideally, since all these calls currently trigger a subscribe (so that therefore ongoing subscribes will delay latter ones), we want to mux the add and remove calls for `updateVideoSourceSubscriptions`. Perhaps we can use a prepare-commit sort of pattern? where we mark sources as 'to-be-subscribed' or 'to-be-not-subscribe' and have some sort of `commit`/`sendUpdates` function on the model to actually call `updateVideoSourceSubscriptions`